Realm/Realm/Realm.cs

```diff
@@ -724,6 +734,30 @@ public bool Refresh()
             return SharedRealmHandle.Refresh();
         }
 
+        /// <summary>
+        /// Waits for the realm and outstanding objects to be updated to a newer version.
+        /// </summary>
+        /// <returns>An awaitable <see cref="Task"/>.</returns>
+        private Task WaitForChangesAsync()
```

Comment:
I hope you don't mind a comment from an outsider. Could you consider making this public? And why the change in name compared to `Refresh`? I would expect it to be called `RefreshAsync`

Reply:
Hmm.. I think I get it. You need to compare the current transaction version number with something, to avoid waiting in vane for a change that may never come. Is it not possible to query the realm for the highest transaction version seen?